RealmError: Incompatible lock file /  After 2.0 RealmFileException  All processes sharing a single file must be the same architecture.

#### Goal

To report specific error after upgrading v0.88.0
#### Expected Results

Crash is not occurred.
#### Actual Results

```
Fatal Exception: io.realm.exceptions.RealmError: Unrecoverable error. Incompatible lock file in io_realm_internal_SharedGroup.cpp line 113
   at io.realm.internal.SharedGroup.createNativeWithImplicitTransactions(SharedGroup.java)
   at io.realm.internal.SharedGroup.(SharedGroup.java:63)
   at io.realm.internal.SharedGroupManager.(SharedGroupManager.java:48)
   at io.realm.BaseRealm.(BaseRealm.java:76)
   at io.realm.Realm.(Realm.java:138)
   at io.realm.Realm.createAndValidate(Realm.java:268)
   at io.realm.Realm.createInstance(Realm.java:248)
   at io.realm.RealmCache.createRealmOrGetFromCache(RealmCache.java:114)
   at io.realm.Realm.getInstance(Realm.java:213)
   at kr.co.vcnc.android.couple.feature.MainTabIndicator.updateHomeBadge(MainTabIndicator.java:138)
   at kr.co.vcnc.android.couple.feature.MainTabIndicator.onAttachedToWindow(MainTabIndicator.java:83)
   at android.view.View.dispatchAttachedToWindow(View.java:12626)
   at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:2473)
   at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:2480)
   at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:2480)
   at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:2480)
   at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:2480)
   at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:2480)
   at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:2480)
   at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:2480)
   at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:2480)
   at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:2480)
   at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:1264)
   at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1047)
   at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:5896)
   at android.view.Choreographer$CallbackRecord.run(Choreographer.java:761)
   at android.view.Choreographer.doCallbacks(Choreographer.java:574)
   at android.view.Choreographer.doFrame(Choreographer.java:544)
   at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:747)
   at android.os.Handler.handleCallback(Handler.java:733)
   at android.os.Handler.dispatchMessage(Handler.java:95)
   at android.os.Looper.loop(Looper.java:136)
   at android.app.ActivityThread.main(ActivityThread.java:5120)
   at java.lang.reflect.Method.invokeNative(Method.java)
   at java.lang.reflect.Method.invoke(Method.java:515)
   at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:792)
   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:608)
   at dalvik.system.NativeStart.main(NativeStart.java)
```
#### Steps & Code to Reproduce

I got this stack trace from Fabric. I've tested several devices that we have, i can't reproduce it.
Instead, i attached some code that error is happening.

FYI, at this update (to v0.88.0), we upgraded Realm's schema version. 
(We're using `deleteRealmIfMigrationNeeded` option)
#### Code Sample

```
@Override
protected void onAttachedToWindow() {
    homeBadgeSubscription = Realm.getInstance(CoupleApplication.getRealmHome())
                .where(RHomeCardViewView.class)
                .equalTo("closed", false)
                .equalTo("newIndicator", true)
                .findAllAsync()
                .asObservable()
                .filter(RealmResults::isLoaded)
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(BasicSubscriber.    <RealmResults<RHomeCardViewView>>create().next(homeCardViewViews -> {
                    int size = homeCardViewViews.size();
                    updateTextView(size);
                }));

@Override
protected void onDetachedFromWindow() {
    if (homeBadgeSubscription != null) {
        homeBadgeSubscription.unsubscribe();
    }
}
```
#### Version of Realm and tooling

Realm version(s): v0.88.0

Which Android version and device: 
When i first saw this crash, i think this is occurred because users are using BlackBerry. 
But now Android users have a crash.

![screen shot 2016-03-15 at 10 55 28 am](https://cloud.githubusercontent.com/assets/1873136/13765402/811f1c0c-ea9c-11e5-9925-451e7411d39a.png)
![screen shot 2016-03-15 at 10 55 40 am](https://cloud.githubusercontent.com/assets/1873136/13765405/839b1a62-ea9c-11e5-81c2-34cc3181686b.png)
![screen shot 2016-03-15 at 10 55 47 am](https://cloud.githubusercontent.com/assets/1873136/13765407/85bebc68-ea9c-11e5-99a3-104287c4a8a1.png)

The peak point is yesterday. (Realm v0.88.0)
![screen shot 2016-03-15 at 10 59 39 am](https://cloud.githubusercontent.com/assets/1873136/13765472/24e53bf0-ea9d-11e5-898e-daa274ed70e0.png)

I also saw the following crash for a user:

Caused by: io.realm.exceptions.RealmError: Could not open the Realm file: Incompatible lock file. Shared info version doesn't match, 259 7.
	at io.realm.internal.SharedGroup.openSharedGroupOrFail(SharedGroup.java:112)
	at io.realm.internal.SharedGroup.<init>(SharedGroup.java:74)
	at io.realm.internal.SharedGroupManager.<init>(SharedGroupManager.java:49)
	at io.realm.BaseRealm.<init>(BaseRealm.java:86)
	at io.realm.Realm.<init>(Realm.java:135)
	at io.realm.Realm.createAndValidate(Realm.java:233)
	at io.realm.Realm.createInstance(Realm.java:214)
	at io.realm.RealmCache.createRealmOrGetFromCache(RealmCache.java:126)
	at io.realm.Realm.getDefaultInstance(Realm.java:160)


Device: Galaxy S4 Mini (serranovelte)  Samsung 
Android-version Android 4.4

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -2,11 +2,14 @@
 
 ### Bug Fixes
 
-* Leaked file handler in the Realm Transformer (#5521)
+* Leaked file handler in the Realm Transformer (#5521).
+* Potential fix for "RealmError: Incompatible lock file" crash (#2459).
 
 ### Internal
 
-* Updated JavaAssist to 3.22.0-GA
+* Updated JavaAssist to 3.22.0-GA.
+* Upgraded to Realm Sync 2.1.4.
+* Upgraded to Realm Core 4.0.3.
 
 
 ## 4.1.1 (2017-10-27)
@@ -19,7 +22,7 @@
 
 ### Internal
 
-* Updated Realm Sync to 2.1.0
+* Updated Realm Sync to 2.1.0.
 
 
 ## 4.1.0 (2017-10-20)

--- a/dependencies.list
+++ b/dependencies.list
@@ -1,7 +1,7 @@
 # Realm Sync Core release used by Realm Java
 # https://github.com/realm/realm-sync/releases
-REALM_SYNC_VERSION=2.1.0
-REALM_SYNC_SHA256=cd52b2ee53ef80b4b9ec80eede7ca5fa28a96353ad7e4d26cf516dbb12586966
+REALM_SYNC_VERSION=2.1.4
+REALM_SYNC_SHA256=6d32ef44acbf4a63b654ceeaadce036feeefd04a4ca649a95a22a0e7d56df84d
 
 # Object Server Release used by Integration tests. Installed using NPM.
 # Use `npm view realm-object-server versions` to get a list of available versions.

